code/game/objects/effects/spiders.dm

```diff
@@ -249,6 +232,27 @@
 	if(amount_grown)
 		amount_grown += rand(0,2)
 
+/obj/effect/spider/spiderling/proc/skitter()
+	if(isturf(loc))
+		if(prob(25))
+			var/list/nearby = trange(5, src) - loc
+			if(nearby.len)
+				var/target_atom = pick(nearby)
+				walk_to(src, target_atom, 5)
+				if(prob(25))
+					src.visible_message("<span class='notice'>\The [src] skitters[pick(" away"," around","")].</span>")
+		else if(prob(5))
+			//vent crawl!
+			for(var/obj/machinery/atmospherics/unary/vent_pump/v in view(7,src))
+				if(!v.welded)
+					entry_vent = v
+					walk_to(src, entry_vent, 5)
+					break
+		if(amount_grown >= 100)

```

Comment:
Not quite sure why this part is in the skitter proc.

Reply:
It was already part of the skitter code, and I didn't want to change it too much from what it was.

Reply:
I think it belongs in the process block, rather than here. Could be wrong.
Did you check that they can still grow up?

Reply:
Nevermind, have been explained why this is here.